.github/workflows/ubuntu.yml

```diff
@@ -171,7 +172,19 @@ jobs:
     - name: Build Pandas
       uses: ./.github/actions/build_pandas
 
-    - name: Test
+    - name: Test (not single_cpu)
       uses: ./.github/actions/run-tests
       # TODO: Don't continue on error for PyPy
       continue-on-error: ${{ env.IS_PYPY == 'true' }}
+      env:
+        # Set pattern to not single_cpu if not already set
+        PATTERN: ${{ env.PATTERN == '' && 'not single_cpu' || matrix.pattern }}
+
+    - name: Test (single_cpu)
```

Comment:
I think we need to ensure this still runs if the previous step fails due to a test failure.

We use `if: ${{ steps.build.outcome == 'success' && always() }}` for that

Reply:
Thanks.